CORE-16983 / CORE-16967 Atomic Swap with Facade

jmacmahonr3 opened the pull request:
API additions in https://github.com/corda/corda-api/pull/1282

szymonsztuka committed:
Jackson serializers and deserializers for DigitalSignatureAndMetadata including all nested types used inside DigitalSignatureAndMetadata and especially proof of batch signature.
 The test using extended UtxoLedgerTest class which produces transaction with a Merkle tree proof - signed by "signBatch".
 Due to quite awkward dependencies between modules, the test is in UTXO module - as only this module can produce a sample signature with proof.

szymonsztuka committed:
Remove unnecessary comment

jmacmahonr3 committed:
Merge branch 'release/interop/syntax' into interop/jmac/CORE-16983

jmacmahonr3 committed:
initial draft pr

vlajos commented on components/flow/flow-service/build.gradle:
```diff
@@ -48,6 +48,7 @@ dependencies {
     implementation project(':libs:platform-info')
     implementation project(":libs:serialization:serialization-avro")
     implementation project(":libs:tracing")
+    implementation project(':libs:crypto:merkle-impl')
```
Depending on `-impl` looks a bit suspicious.
Dependencies should be injected via OSGi.
Although probably some direct constructors are not available via OSGi.

jmacmahonr3 committed:
Now adding test cordapp

jmacmahonr3 committed:
Update to serializer

szymonsztuka committed:
Add new serializers and deserializers to main Jackson service (for now).

szymonsztuka committed:
1. Create unit test to recreate: java.lang.ClassCastException: class com.fasterxml.jackson.databind.MappingIterator cannot be cast to class java.lang.String
2. Move addition of new Jackson deserializers to initialisation of JsonMarshallingServiceImpl, instead adding them in a pretty much random method "setDeserializer" which may be not invoked

szymonsztuka committed:
Serializing and deserializing Proof Of action solved.

szymonsztuka committed:
Key creation from bytes solved.

szymonsztuka committed:
Remove test class is the extended version is now in the other module.

szymonsztuka committed:
Test flow (unrealistic one) to check signature verification.

szymonsztuka committed:
Update api version to a temporary build, add TODOs

szymonsztuka committed:
Merge branch 'release/interop/syntax' into interop/jmac/CORE-16983

szymonsztuka committed:
Remove unnecessary function with "default" parameter.

szymonsztuka committed:
Remove unnecessary function with "default" parameter use by tests only.

szymonsztuka committed:
Facade Specification doesn't require new serializers, only producing a request/response does.

szymonsztuka committed:
FacadeRequestSerializer / FacadeResponseSerializer don't need the explicit inner assignment of new ProofOfAction serialization module, as it will be "injected" from the "main" serializer.
Remove unnecessary introduced  code differences to the syntax branch.

szymonsztuka committed:
Remove unused change in a dependency.

szymonsztuka committed:
Verify Signature via RPC flow call.

szymonsztuka committed:
Merge branch 'release/interop/syntax' into interop/jmac/CORE-16983

szymonsztuka committed:
Update LockState and Contract to use DigitalSignatureAndMetadata.

szymonsztuka committed:
Enable serialisation of PublicKey.

szymonsztuka committed:
Query to check owned states.

szymonsztuka committed:
Swap - part on the transaction draft side.

szymonsztuka committed:
Merge branch 'release/interop/syntax' into interop/jmac/CORE-16983

# Conflicts:
#	gradle.properties
#	testing/cpbs/atomic-swap-contracts/src/main/kotlin/com/r3/corda/atomic/swap/contracts/LockContract.kt
#	testing/cpbs/atomic-swap-workflows/src/main/kotlin/com/r3/corda/atomic/swap/workflows/CreateLockFlow.kt

szymonsztuka committed:
Fix compilation issues after merge from syntax branch.

szymonsztuka committed:
Swap - part on transaction with a lock side.

szymonsztuka committed:
Swap - facade method lock with a generic asset ID instead of a more specific amount and denomination parameters.

szymonsztuka committed:
DTOs returned by Query Flows.

szymonsztuka committed:
Move new additions to  UtxoLedgerTest for now into interop test.

szymonsztuka committed:
Test signature proof generation by a notary.

szymonsztuka committed:
Test signature proof generation by a notary.

szymonsztuka committed:
Merge branch 'szymon/interop/add-signing-utxo-derived-test' into interop/jmac/CORE-16983

# Conflicts:
#	components/flow/flow-service/build.gradle
#	components/flow/flow-service/src/main/kotlin/net/corda/flow/application/services/impl/FacadeServiceImpl.kt
#	components/flow/flow-service/src/test/kotlin/net/corda/flow/application/services/interop/dispatchers/FacadeServerDispatcherSpec.kt
#	components/flow/flow-service/src/test/kotlin/net/corda/flow/application/services/interop/proxies/FacadeClientProxySpec.kt
#	components/flow/flow-service/src/test/kotlin/net/corda/flow/application/services/interop/roundtrip/RoundTripTest.kt
#	gradle.properties
#	testing/cpbs/interop-token-demo-workflows/src/main/kotlin/com/r3/corda/demo/interop/tokens/workflows/interop/SwapResponderSubFlow.kt

szymonsztuka committed:
Merge branch 'release/interop/syntax' into interop/jmac/CORE-16983

szymonsztuka committed:
Updates after merge from syntax.

szymonsztuka committed:
Updates after merge from syntax.

szymonsztuka committed:
Updates after merge from syntax.

szymonsztuka committed:
Merge branch 'release/interop/syntax' into interop/jmac/CORE-16983

# Conflicts:
#	components/ledger/ledger-utxo-flow/src/test/kotlin/net/corda/ledger/utxo/flow/impl/transaction/signatureproof/SignatureProofTests.kt
#	testing/ledger/ledger-utxo-base-test/src/main/kotlin/net/corda/ledger/utxo/test/UtxoLedgerWithBatchSignerTest.kt

szymonsztuka committed:
Remove dependency on implementation library by Json Service, use service injection.

szymonsztuka committed:
Separate service

szymonsztuka committed:
Separate service

wzur-r3 reviewed (approved):
LGTM
